src/parser/ExpressionParser.mts

```diff
@@ -1465,16 +1464,23 @@ export abstract class ExpressionParser extends FunctionParser {
                    && this.scope.hasSuperCall(),
       }, () => {
         const body = this.parseFunctionBody(isAsync, isGenerator, false);
-        // NOTE: since the property name below is a union, it is unsound to write to `node` in this fashion
-        // @ts-expect-error
-        node[`${isAsync ? 'Async' : ''}${isGenerator ? 'Generator' : 'Function'}Body`] = body;
+        // Unsafe cast below
```

Comment:
I'd been thinking of handling this using generics and callbacks to avoid the unsound cast.